Source/WebCore/dom/ContentVisibilityDocumentState.cpp

```diff
@@ -230,4 +232,23 @@ void ContentVisibilityDocumentState::removeViewportProximity(const Element& elem
     m_elementViewportProximities.remove(element);
 }
 
+void ContentVisibilityDocumentState::elementContentVisibilityChange(const Element& element, const ContentVisibility& oldValue, const ContentVisibility& newValue)
+{
+    ASSERT(oldValue != newValue);
+    if (oldValue == newValue || newValue != ContentVisibility::Visible)
+        return;
+    for (auto* animation : WebAnimation::instances()) {
```

Comment:
If the change happens per document, I feel like this should go through animations registered for this document alone instead of all animation instances.

Reply:
The new PR uses matchingAnimations.

Reply:
Reverted that as discussed.